code/game/objects/items/devices/radio/listening_bugs.dm

```diff
@@ -0,0 +1,302 @@
+#define DISGUISE_REMOVE "remove disguise"
+#define DISGUISE_RADIO "radio"
+#define DISGUISE_PEN "pen"
+#define DISGUISE_FOUNTAIN_PEN "fountain pen"
+#define DISGUISE_ACCESS_TUNER "access tuner"
+#define DISGUISE_WHISTLE "whistle"
+#define DISGUISE_MASS_SPEC "mass-spectrometer"
+#define DISGUISE_CAMERA "camera"
+#define DISGUISE_ZIPPO "zippo lighter"
+#define DISGUISE_TAPE_RECORDER "tape recorder"
+
+/obj/item/device/radio/listening_bug
+	name = "listening device"
+	desc = "A small, and disguisable, listening device."
+
+	icon = 'icons/obj/items/devices.dmi'
+	icon_state = "voice0"
+	item_state = "analyzer"
+
+	w_class = SIZE_TINY
+	volume = RADIO_VOLUME_RAISED
+
+	broadcasting = FALSE
+	listening = FALSE
+	frequency = BUG_A_FREQ
+	canhear_range = 2
+	freqlock = TRUE
+	/// What the bug is currently pretending to be.
+	var/current_disguise = DISGUISE_REMOVE
+	/// Whether or not the bug can be used to listen to its own channel.
+	var/prevent_snooping = FALSE
+
+/obj/item/device/radio/listening_bug/ui_data(mob/user)
+	var/list/data = list()
+
+	data["broadcasting"] = broadcasting
+	data["listening"] = listening
+	data["frequency"] = frequency
+	data["freqlock"] = freqlock
+
+	var/list/radio_channels = list()
+
+	for(var/channel in channels)
+		var/channel_key = channel_to_prefix(channel)
+		radio_channels += list(list(
+			"name" = channel,
+			"status" = channels[channel] & FREQ_LISTENING,
+			"hotkey" = channel_key))
+
+	data["channels"] = radio_channels
+
+	data["command"] = volume
+	data["useCommand"] = use_volume
+	data["subspace"] = subspace_transmission
+	data["subspaceSwitchable"] = subspace_switchable
+	data["headset"] = FALSE
+
+	return data
+
+/obj/item/device/radio/listening_bug/ui_act(action, params, datum/tgui/ui, datum/ui_state/state)
+	switch(action)
+		if("listen")
+			if(prevent_snooping)
+				to_chat(usr, SPAN_WARNING("This device cannot receive transmissions!"))
+				return
+			listening = !listening
+			return
+		if("subspace")
+			if(!ishuman(usr))
+				return
+			var/mob/living/carbon/human/user = usr
+			if(!check_access(user.wear_id) && !check_access(user.get_active_hand()))
+				to_chat(user, SPAN_WARNING("You need an authenticated ID card to change this function!"))
+				return
+			if(subspace_switchable)
+				subspace_transmission = !subspace_transmission
+				var/initial_prevent = initial(prevent_snooping)
+				if(initial_prevent)
+					prevent_snooping = TRUE
+				if(!subspace_transmission)
+					prevent_snooping = FALSE
+					channels = list()
+				return
+	..()
+
+/obj/item/device/radio/listening_bug/hear_talk(mob/M as mob, msg, verb = "says", datum/language/speaking = null)
+	var/processed_verb = "[SPAN_RED("\[LSTN Device\]")] [verb]"
+	if(broadcasting)
+		if(get_dist(src, M) <= 7)
```

Comment:
why check dist? if it hear_talk it hear_talk, range checks have already been done

Reply:
Have to override whatever is going on in /obj/itm/device/radio

`/obj/item/device/radio/hear_talk(mob/M as mob, msg, verb = "says", datum/language/speaking = null)
	if (broadcasting)
		if(get_dist(src, M) <= canhear_range)
			talk_into(M, msg,null,verb,speaking)`